### PR TITLE
feat(gui-client): Tauri welcome screen

### LINKED ISSUE #3961: Don't show Settings window on App Start

### PR DESCRIPTION
**Pull request by ReactorScram**
Closes #3961 

No tests yet, might be tricky to test since it's all I/O. 
I cued it off the device ID being generated, so it will have a minor merge conflict with #3920

```[tasklist]
### Before merging
- [ ] UI polish, or disable the welcome screen temporarily
```

image

### CODE REVIEW

File: rust/gui-client/src-tauri/src/client/gui.rs [hunk 1]
```diff
@@ -791,6 +791,14 @@ async fn run_controller(
         .await
         .context("Failed to read / create device ID")?;
 
+    if device_id.generated {
```
Comment (jamilbk) [inline rust/gui-client/src-tauri/src/client/gui.rs]: I'm probably missing something, but wouldn't this be `if !device_id.generated`?

Comment (ReactorScram) [reply rust/gui-client/src-tauri/src/client/gui.rs]: I guess it's ambiguous between "generated ever" and "generated just now"

If I change the variable name to `generated_just_now` or maybe `first_time` would that clear it up?

Comment (jamilbk) [reply rust/gui-client/src-tauri/src/client/gui.rs]: Yep! I had the same thought.

Comment (jamilbk) [reply rust/gui-client/src-tauri/src/client/gui.rs]: Yep! I had the same thought.

**Comment by jamilbk** [comment]
LGTM, OK if I push a couple UI commits into this PR at some point?

**Comment by ReactorScram** [comment]
@jamilbk Yep go for it

File: rust/gui-client/src-tauri/src/client/gui.rs (hunk 1 above)
Comment (ReactorScram) [reply rust/gui-client/src-tauri/src/client/gui.rs]: Changed it to `is_first_time`

**Comment by jamilbk** [comment]
Still need to get to this one.

**Comment by ReactorScram** [comment]
I could reference the About and Settings windows and just make something rough for the time being

**Comment by jamilbk** [comment]
@ReactorScram I think this is ready for you take over. I need help wiring up the Sign in button to close the welcome window and launch the sign in flow. Could you finish that part of this PR? I think it's ready for merge after that.

Screenshot 2024-03-17 at 9 01 19 PM

**Comment by ReactorScram** [comment]
Got it hooked up and it works fine in Windows and Debian. 

It sends the same command as clicking "Sign In" in the tray menu, so both of them will hide the Welcome window and open the browser for auth. If the window's already hidden it just opens the browser, since `hide` is idempotent.

image

**Review by jamilbk — APPROVED** [review]

**Pull request merged**